### DISCUSSION TITLE
4 pictures into one, many functions are quoted

### DISCUSSION THREAD

**Discussion by s01o** [ImageMagick/ImageMagick#4244] [Help]
[https://i.ibb.co/B6s1YV0/rr1.png](https://i.ibb.co/B6s1YV0/rr1.png)

Can the admin help me? Combine these steps into one command,
that 4img is same Width&Height and quote files in 
four folders [in1\00000.png-09999.png][in1\00000.png-09999.png][in3\ 00000.png-09999.png][in\00000.png-09999.png], 
the output file is [out\00000.png-09999.png], 
the degree of lightening and darkening can be similar to the figure . 
grateful 

Translated from GOOGLE, please forgive me if there is a semantic barrier. 

I can only use CMD in WIN10 to operate IM  , version 7.1.0-7

**Comment by snibgo** [comment]
The question is vague and seems to have many parts. Please break it down. What is the first problem you need to solve?

**Reply by s01o** [reply]
Thanks for your recovery ，

-----
![00000](https://user-images.githubusercontent.com/26829509/133910622-9d724a16-121e-439c-a549-4fd3117b85be.png)
layer1\00000.png
-----
![00000](https://user-images.githubusercontent.com/26829509/133910627-91138292-8bbc-4d35-98f0-971a2f5fcf0b.png)
layer2\00000.png
-----
![00000](https://user-images.githubusercontent.com/26829509/133910632-7450bcd4-f34b-447a-8183-3a7a737e77e1.png)
layer3\00000.png
-----
![00000](https://user-images.githubusercontent.com/26829509/133910634-ab4001e7-d048-44bd-854f-38427d1b012f.png)
layer4\00000.png


In all steps, the top and bottom order of the images is 
layer1\00000.png
layer2\00000.png
layer3\00000.png
layer4\00000.png

1
![1](https://user-images.githubusercontent.com/26829509/133911023-f7c538e2-8499-46d0-90fe-eae9941ab26e.png)
layer2\00000.png & layer3\00000.png
get the common part → step1.png

2
![2](https://user-images.githubusercontent.com/26829509/133911028-e776da3e-7c1d-4694-ae80-530c3e429133.png)
step1.png & layer3\00000.png
blend mode-multiply → step2.png

3
![3](https://user-images.githubusercontent.com/26829509/133911029-9542a274-06b5-4325-bff4-003ea650a008.png)
layer2\00000.png - layer3\00000.png → step3.png

4
![4](https://user-images.githubusercontent.com/26829509/133911030-06b6185a-fcb1-424a-9ea6-28d4d7087e52.png)
step1_3.png remove white background  → step4.png

5
![5](https://user-images.githubusercontent.com/26829509/133911032-fd514598-c030-42fa-941e-0b7bb2682eec.png)
step4.png darken  → step1_5.png

6
![6](https://user-images.githubusercontent.com/26829509/133911034-fe4bebb5-3a84-4d13-bf5c-31b2fea34379.png)
step2.png + step5.png → step6.png

7
![7](https://user-images.githubusercontent.com/26829509/133911038-08975af3-42d5-4eed-ba4f-4aea6746225b.png)
step6.png Remove all areas with transparency greater than 0  → step7.png

8
![8](https://user-images.githubusercontent.com/26829509/133911070-82936c09-b424-4e9c-aa89-3824742a6d72.png)
layer1\00000.png & step7.png
get the common part → step8.png

9
![9](https://user-images.githubusercontent.com/26829509/133911074-37b2990c-50c8-4f01-9059-6b4117bbcf6b.png)
step7.png & step8.png
blend mode-screen → step9.png

10
![10](https://user-images.githubusercontent.com/26829509/133911076-79fd5974-836a-4360-a252-e9001615b374.png)
layer1\00000.png - step8.png → step10.png

11
![11](https://user-images.githubusercontent.com/26829509/133911077-52f83556-5667-4c21-91ef-2155676b081e.png)
step10.png remove black background  → step11.png

12
![12](https://user-images.githubusercontent.com/26829509/133911078-f27d6052-aa40-4a79-82c1-ed9fee0be7ea.png)
step11.png brighten → step12.png
I'm not sure this way can be used to make this effect closer
img is overlaid on a dark background with screenblendmode
if you have one,please help me

13
![13](https://user-images.githubusercontent.com/26829509/133911081-aafa154f-1b90-4f56-84ff-9979aa6dd74a.png)
step6.png - step7.png → step13.png

14
![14](https://user-images.githubusercontent.com/26829509/133911082-a65f1a50-1e70-4daf-904e-42e4c1fd784f.png)
layer4\00000.png transparency 60% → step14.png

15
![15](https://user-images.githubusercontent.com/26829509/133911084-56c3f717-f672-4946-9add-0fd6fa7c5593.png)
step12.png + step13.png + step9.png + step14.png → out\00000.png

**Comment by fmw42** [comment]
Do you have a Photoshop or GIMP file showing what you want to achieve? Perhaps reviewing the layers in the file will make more sense to someone.

**Reply by s01o** [reply]
[41a.zip](https://github.com/ImageMagick/ImageMagick/files/7190778/41a.zip)
grateful

The game material itself has four layers
But I need to make a webpage database. If CSS is used for overlay, it will take up more computer resources, so I want to make a separate picture.

**Comment by fmw42** [comment]
Please post another file with only the layers you really want. I see multiple layers marked 1,2,3,4. I need to see which ones you really need and separate them as PNG files so I can composite in ImageMagick. I need to look at each one to see it s blend modes and transparency settings.

**Reply by fmw42** [reply]
Are you needing to export the layers as PNGs and work on them in ImageMagick or do you just have original PNGs or TIFFs or JPGs that have no blend and no transparency associated or do they already have the appropriate transparencies?  What exactly are you using for input? If you have them as separate and not too large files, please post those.

**Reply by s01o** [reply]
[imput.zip](https://github.com/ImageMagick/ImageMagick/files/7190825/imput.zip)
Sorry, the img import description I described above is wrong, the imgs are all in the same folder, and only the first number is different on each level.

1abcd.png,2abcd.png,3abcd.png,4abcd.png 👉 0abcd.png 

10352.png,20352.png,30352.png,40352.png 👉 00352.png
10353.png,20353.png,30353.png,40353.png 👉 00353.png

**Comment by fmw42** [comment]
You have a lot of png files. Please identify a list of each step you want. Such as:

```
1) Image1 --- background
2) Image2 onto Image1 using blending mode X and Opacity Y%
3) etc

```

Can we limit the discussion to one set. Then you can understand and do all your other sets? I assume this

1abcd.png,2abcd.png,3abcd.png,4abcd.png 👉 0abcd.png

is one set and that the same thing is applied with sets 

10352.png,20352.png,30352.png,40352.png 👉 00352.png
10353.png,20353.png,30353.png,40353.png 👉 00353.png

So if I show you for the first 4 images, can you then do the rest that you want?

**Reply by s01o** [reply]
Yes, that's right
There is only a problem with the input file just now, the other steps are just as I described above.

**Reply by s01o** [reply]
Now steps 2 and 9 are blending mode multiply and screen
The steps I’m not sure about are 5 and 12. In what way to brighten and darken, and how much is the amount, you can set a value for me first, and I will adjust it later. grateful!

**Reply by fmw42** [reply]
Sorry, I do not understand your diagrams/images. So that is why I asked you to provide a list of steps as I specified.  Tell me what you want to do just for steps 2 and then just for steps 9. That is what blending mode, what opacity, what brightening or anything else that needs to be done for each of those two steps. Do you have masks that need to be used as well to control opacity or brightening?

If you can provide a simple Photoshop file with just step 2 and then another with just step 9 on some simple images, then perhaps I can look at what you are doing. But the PS file you provided before was way too complicated for me to follow.

**Reply by fmw42** [reply]
In ImageMagick, if you want to composite two image with some blending mode, you do that as follows. (See https://legacy.imagemagick.org/Usage/layers/#convert and https://legacy.imagemagick.org/Usage/compose/#compose )

`convert background.png overlay.png -compose blending_mode -composite result.png
`

or if you need to use a mask to control where the blending is done, you add that as third input as:

`convert background.png overlay.png mask.png -compose blending_mode -composite result.png
`
So if multiply

`convert background.png overlay.png -compose multiply -composite result.png
`
Or if screen

`convert background.png overlay.png -compose screen -composite result.png

If you want multiply and some less than 100% opaque, say, 70% opaque, then you use parentheses to separate the processing on the overlay image and select the alpha channel and set it to 70%. So you are basically doing two converts in one, by using the parentheses (without the second convert). The first is inside the parentheses, but unspecified, and applies modification to the overlay image. Then the result of that is applied by the main convert to the background to do the blending according to the blending method.

`convert  background.png ( overlay.png -channel a -evaluate set 70% +channel ) -compose multiply -composite result.png`
`

If you want multiply blending and some brightness adjustment, there are many adjustments that you can use. A simple linear one is -level b,w%  for blackpoint and whitepoint percent. You can use -gamma X or -sigmoidal-contrast Y. Or you can set up a curved mapping by building a look-up table and applying that to the image with -clut.  Please review those methods at 

http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=9620
http://www.imagemagick.org/script/command-line-processing.php
https://imagemagick.org/script/command-line-options.php
http://www.imagemagick.org/Usage/reference.html
http://www.imagemagick.org/Usage/


So for example if you want a brightness-contrast adjustment you could do

`convert  background.png ( overlay.png -brightness-contrast 20,10 ) -compose multiply -composite result.png`
`

The above is DOS Windows syntax. For Unix-like systems escape the parentheses with \ so \\( ... \\).
For ImageMagick 7, replace convert with magick.


Suppose you want to composite image2 onto image1 with multiply, then composite image3 onto the result of the first composite using blending of screen. That would be

`convert image1 image2 -compose multiply -composite image3 -compose screen -composite result
`

The result of the first composite is implicit and not specified
